Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-187.ca-central-1.compute.internal
 Git:  git version 2.9.5
Sha: be01358542eb7aace2088aa3a5ee741aae280dbc
Number of Error(s): 5
/home/centos/smoketest/PR-15033/HPCC-Platform/system/jlib/jstats.cpp:1563:13: error: \xe2\x80\x98setFileId\xe2\x80\x99 was not declared in this scope
/home/centos/smoketest/PR-15033/HPCC-Platform/system/jlib/jstats.cpp:1631:48: error: no \xe2\x80\x98void StatsScopeId::setFileId(const char*)\xe2\x80\x99 member function declared in class \xe2\x80\x98StatsScopeId\xe2\x80\x99
checking for libdl (by compile)...make[2]: *** [system/jlib/CMakeFiles/jlib.dir/build.make:570: system/jlib/CMakeFiles/jlib.dir/jstats.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1302: system/jlib/CMakeFiles/jlib.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2021-06-09_16-55-52
time logs started
Prep time   :     9 sec (00:00:09)
Build time  :   134 sec (00:02:14)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   143 sec (00:02:23)
time logs end

[Build error:](https://gist.github.com/e0dbb9b4babeb9226d97b814b32582b1)
Build log [Build log](https://gist.github.com/HPCCSmoketest/e0dbb9b4babeb9226d97b814b32582b1#file-result-21-06-09-16-53-16-log)

--- a/system/jlib/jstatcodes.h
+++ b/system/jlib/jstatcodes.h
@@ -26,6 +26,7 @@
 #define FunctionScopePrefix "f"
 #define WorkflowScopePrefix "w"
 #define ChildGraphScopePrefix "c"
+#define FileScopePrefix "p"
 
 #define MATCHES_CONST_PREFIX(search, prefix) (strncmp(search, prefix, strlen(prefix)) == 0)
 
@@ -69,6 +70,7 @@ enum StatisticScopeType
     SSTfunction,                        // a function call
     SSTworkflow,
     SSTchildgraph,
+    SSTfile,
     SSTunknown,
     SSTmax
 };

--- a/system/jlib/jstats.cpp
+++ b/system/jlib/jstats.cpp
@@ -72,7 +72,7 @@ void setStatisticsComponentName(StatisticCreatorType processType, const char * p
 // Textual forms of the different enumerations, first items are for none and all.
 static constexpr const char * const measureNames[] = { "", "all", "ns", "ts", "cnt", "sz", "cpu", "skw", "node", "ppm", "ip", "cy", "en", "txt", "bool", "id", "fname", "cost", NULL };
 static constexpr const char * const creatorTypeNames[]= { "", "all", "unknown", "hthor", "roxie", "roxie:s", "thor", "thor:m", "thor:s", "eclcc", "esp", "summary", NULL };
-static constexpr const char * const scopeTypeNames[] = { "", "all", "global", "graph", "subgraph", "activity", "allocator", "section", "compile", "dfu", "edge", "function", "workflow", "child", "unknown", nullptr };
+static constexpr const char * const scopeTypeNames[] = { "", "all", "global", "graph", "subgraph", "activity", "allocator", "section", "compile", "dfu", "edge", "function", "workflow", "child", "file", "unknown", nullptr };
 
 static unsigned matchString(const char * const * names, const char * search, unsigned dft)
 {
@@ -111,6 +111,7 @@ static const StatisticScopeType scoreOrder[] = {
     SSTfunction,
     SSTworkflow,
     SSTchildgraph,
+    SSTfile,
     SSTunknown
 };
 static int scopePriority[SSTmax];
@@ -1408,6 +1409,7 @@ void StatsScopeId::describe(StringBuffer & description) const
     case SSTedge:
         description.append(' ').append(id).append(',').append(extra);
         break;
+    case SSTfile:
     case SSTfunction:
         description.append(' ').append(name);
         break;
@@ -1479,6 +1481,7 @@ void StatsScopeId::serialize(MemoryBuffer & out) const
         out.append(id);
         out.append(extra);
         break;
+    case SSTfile:
     case SSTfunction:
         out.append(name);
         break;
@@ -1554,6 +1557,15 @@ bool StatsScopeId::setScopeText(const char * text, const char * * _next)
             return true;
         }
         break;
+    case FileScopePrefix[0]:
+        if (MATCHES_CONST_PREFIX(text, FileScopePrefix))
+        {
+            setFileId(text+strlen(FileScopePrefix));
+            if (_next)
+                *_next = text + strlen(text);
+            return true;
+        }
+        break;
     case WorkflowScopePrefix[0]:
         if (MATCHES_CONST_PREFIX(text, WorkflowScopePrefix) && isdigit(text[strlen(WorkflowScopePrefix)]))
         {
@@ -1614,6 +1626,11 @@ void StatsScopeId::setSubgraphId(unsigned _id)
 void StatsScopeId::setFunctionId(const char * _name)
 {
     scopeType = SSTfunction;
+    name.set(_name);
+}
+void StatsScopeId::setFileId(const char * _name)
+{
+    scopeType = SSTfile;
     name.set(_name);
 }
 void StatsScopeId::setWorkflowId(unsigned _id)

--- a/system/jlib/jstats.h
+++ b/system/jlib/jstats.h
@@ -82,6 +82,7 @@ public:
     void setActivityId(unsigned _id);
     void setEdgeId(unsigned _id, unsigned _output);
     void setFunctionId(const char * _name);
+    void setFileId(const char * _name);
     void setSubgraphId(unsigned _id);
     void setWorkflowId(unsigned _id);
     void setChildGraphId(unsigned _id);

